### ISSUE TITLE
Port 1.x influxd backup and restore commands

### ISSUE THREAD

**Issue by stuartcarnie** [influxdata/influxdb#19316]
## What

Users will require backup and restore capabilities. Port the TSM backup and restore commands from 1.x and integrate with 2.x.

- [ ] influx backup
- [ ] influx restore

### 2.0 Backup (prior to TSM 1.x transition)

The 2.0 backup / restore command can be summarized as the following:

* Online backup and *offline* restore
* Backup all meta data 
* Backup all buckets

This approach is limited and largely governed by the storage architecture, best described as a single database and single shard. All buckets were co-mingled in a single set of TSM files.

### 2.0 Backup

The OSS 1.8 backup system is considerably more powerful and flexible per [the documentation](https://docs.influxdata.com/influxdb/v1.8/administration/backup_and_restore/).

With the transition to the TSM 1.x storage engine, the on-disk structure lends itself to the same backup / restore features. This layout is discussed in #19114, however, it is included below for reference:

```
data/
  bucket-a/
    autogen/
      <shardid-1>/
        *.tsm
      <shardid-2>/
        *.tsm
  bucket-b/
    autogen/
      <shardid-1>/
        *.tsm
```

In summary, each bucket is now represented as a single database and retention policy and is made up of one or more shards.

Given that, many of the features of 1.8 can be ported, including

* Online backup and restore
* Backup of individual buckets (database / retention policy in 1.x parlance)
* Backup of individual shards (single shard from a single bucket)

**Comment by stephanie-engel** [comment]
Once this fix is implemented, the litmus backup and restore tests will need to be un-skipped.

**Comment by benbjohnson** [comment]
Update: Backup/restore PR is up: https://github.com/influxdata/influxdb/pull/19864

**Comment by benbjohnson** [comment]
Completed & merged into master.
